numba/_helperlib.c

```diff
@@ -275,6 +275,12 @@ numba_recreate_record(void *pdata, int size, PyObject *dtype) {
     PyObject *record = NULL;
     PyArray_Descr *descr = NULL;
 
+    if (dtype == NULL) {
+        PyErr_Format(PyExc_ValueError,
+            "In 'numba_recreate_record', 'descr' is NULL");
+        return NULL;
+    }
```

Comment:
Should this be a RuntimeError (as settled previously elsewhere) also, it refers to `descr` but is checking `dtype`.